src/Microsoft.DocAsCode.Build.Engine/Incrementals/DependencyGraph.cs

```diff
@@ -3,61 +3,61 @@
 
 namespace Microsoft.DocAsCode.Build.Engine.Incrementals
 {
+    using System;
     using System.Collections.Generic;
-    using System.Linq;
     using System.IO;
 
     using Microsoft.DocAsCode.Common;
 
-    public class DependencyGraph
+    public sealed class DependencyGraph
     {
         private readonly HashSet<DependencyItem> _dependencyItems;
+        private readonly Dictionary<string, DependencyType> _types;
         private readonly Dictionary<string, HashSet<DependencyItem>> _indexOnFrom = new Dictionary<string, HashSet<DependencyItem>>();
         private readonly Dictionary<string, HashSet<DependencyItem>> _indexOnReportedBy = new Dictionary<string, HashSet<DependencyItem>>();
-        private static readonly Dictionary<string, DependencyType> _types = new Dictionary<string, DependencyType>();
+        private static readonly Dictionary<string, DependencyType> _defaultTypes = new Dictionary<string, DependencyType>
+        {
+            { DependencyTypeName.Uid, new DependencyType { Name = DependencyTypeName.Uid, IsTransitive = false, TriggerBuild = false } },
+        };
 
-        public DependencyGraph()
-            : this(new HashSet<DependencyItem>())
+        internal DependencyGraph()
+            : this(new HashSet<DependencyItem>(), _defaultTypes)
```

Comment:
nice catch!

---
In reply to: [78122892](https://github.com/dotnet/docfx/pull/681#discussion_r78122892) [](ancestors = 78122892)